File: RepoTransBench/source_projects/Java/JamesZBL_java_design_patterns/composite/src/test/java/me/zbl/composite/WriterIntegrationPublicTest.java
package me.zbl.composite;

import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

class WriterIntegrationPublicTest {
    @Test
    void testWriterSentencesPublic() {
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        PrintStream old = System.out;
        System.setOut(new PrintStream(baos));
        // Call only one sentence to test different output path
        new Writer().sentenceByEnglish().print();
        System.setOut(old);
        String result = baos.toString();
        // Ensure at least the result contains 'student' and a 'from'
        assert(result.contains("student"));
        assert(result.contains("from"));
    }
}

File: RepoTransBench/target_projects/Java/C#/JamesZBL_java_design_patterns/public_tests/WriterIntegrationPublicTest.cs
using System;
using System.IO;
using Xunit;

namespace PublicTests
{
    public class WriterIntegrationPublicTest
    {
        [Fact]
        public void TestWriterSentencesPublic()
        {
            var output = new StringWriter();
            var oldOut = Console.Out;
            Console.SetOut(output);
            new Writer().SentenceByEnglish().Print();
            Console.SetOut(oldOut);
            string result = output.ToString();
            Assert.Contains("student", result);
            Assert.Contains("from", result);
        }
    }
}